### ISSUE TITLE
An error occurred while `Style/ConditionalAssignment` with `EnforcedStyle: assign_inside_condition`

### ISSUE THREAD

**Issue by naft-a** [rubocop/rubocop#10737]
## Expected behavior

Rubocop linting should pass with no errors.

## Actual behavior

Rubocop raises the error `An error occurred while Style/ConditionalAssignment cop was inspecting` and 
`rubocop --debug` did not provide any useful info.

## Steps to reproduce the problem

Error is raised for the line `failures += 1` in this method:

```ruby
    def perform(action = nil)
      failures = 0

      @collection.each do |object|
        yield(object)
      rescue ActiveRecord::RecordInvalid
        failures += 1
      end

      redirect_to request.referer || @return_path, notice: notice_message(@collection.length, failures, action)
    end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.27.0 (using Parser 3.1.1.0, rubocop-ast 1.16.0, running on ruby 3.0.2 x86_64-linux)
  - rubocop-performance 1.13.3
  - rubocop-rails 2.14.2
  - rubocop-rspec 2.9.0
```
